src/auth/index.js

```diff
@@ -169,17 +169,147 @@ export default class Auth {
+   * @param {Object} parameters passwordless parameters
+   * @param {String} parameters.phoneNumber the phone number to send the link/code to
+   * @param {String} parameters.send the passwordless strategy, either 'link' or 'code'
+   * @param {String} parameters.authParams optional parameters, used when strategy is 'linḱ'
+   * @returns {Promise}
+   *
+   * @memberof Auth
+   */
+  passwordlessWithSMS(parameters = {}) {
+    const payload = apply(
+      {
+        parameters: {
+          phoneNumber: {required: true, toName: 'phone_number'},
+          send: {required: false},
+          authParams: {required: false},
+        },
+      },
+      parameters,
+    );
+    return this.client
+      .post('/passwordless/start', {
+        ...payload,
+        connection: 'sms',
+        client_id: this.clientId,
+      })
+      .then(responseHandler);
+  }
+
+  /**
+   * Finishes the Passworldess authentication with an email connection
+   *
+   * @param {Object} parameters passwordless parameters
+   * @param {String} parameters.email the email where the link/code was received
+   * @param {String} parameters.code the code numeric value (OTP)
+   * @param {String} parameters.audience optional API audience to request
+   * @param {String} parameters.scope optional scopes to request
+   * @returns {Promise}
+   *
+   * @memberof Auth
+   */
+  loginWithEmail(parameters = {}) {
+    const payload = apply(
+      {
+        parameters: {
+          email: {required: true, toName: 'username'},
+          code: {required: true, toName: 'otp'},
+          audience: {required: false},
+          scope: {required: false},
+        },
+      },
+      parameters,
+    );
+    return this.client
+      .post('/oauth/token', {
+        ...payload,
+        client_id: this.clientId,
+        realm: 'email',
+        grant_type: 'http://auth0.com/oauth/grant-type/passwordless/otp',
+      })
+      .then(responseHandler);
+  }
+
+  /**
+   * Finishes the Passworldess authentication with an SMS connection
+   *
+   * @param {Object} parameters passwordless parameters
+   * @param {String} parameters.phoneNumber the phone number where the link/code was received
+   * @param {String} parameters.code the code numeric value (OTP)
+   * @param {String} parameters.audience optional API audience to request
+   * @param {String} parameters.scope optional scopes to request
+   * @returns {Promise}
+   *
+   * @memberof Auth
+   */
+  loginWithSMS(parameters = {}) {
+    const payload = apply(
+      {
+        parameters: {
+          phoneNumber: {required: true, toName: 'username'},
+          code: {required: true, toName: 'otp'},
+          audience: {required: false},
+          scope: {required: false},
+        },
+      },
+      parameters,
+    );
+    return this.client
+      .post('/oauth/token', {
+        ...payload,
+        client_id: this.clientId,
+        realm: 'sms',
+        grant_type: 'http://auth0.com/oauth/grant-type/passwordless/otp',
```

Comment:
You might want to extract this, to avoid repeating it in https://github.com/auth0/react-native-auth0/pull/270/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md#diff-2b0a2893c010a4aea8147a48a8fdd092R278

Reply:
This particular endpoint is called for many different flows (5 calls on this file currently). What mostly changes all the time is the grant_type sent. But there are also additional parameters, such as connection/realm, and the additional values the user could include. I'm afraid that moving it into a function would maintain the same structure, as we need to pass those arguments anyway. Worth to revisit later.

Reply:
The way I've seen that Auth0.swift solves it, is to create a custom matcher (a function, in this case) for every use case instead of every endpoint. So it has many matchers for `/oauth/token`, each one configured differently for each use case.

Reply:
And each of these matchers may or may not take params, according to the needs of the use case.